src/main/java/recipeapplication/services/RecipeServiceImpl.java

```diff
@@ -43,4 +41,16 @@ public void addRecipe(Recipe recipe, MultipartFile recipeImage) {
 
 	}
 
+	public Recipe readRecipe(String recipeId) {
+		if (Integer.parseInt(recipeId) > 0)
+			return this.recipes.read(recipeId);
+		else
+			return null;
```

Comment:
As mentioned before, better to throw an exception than returning null values

Reply:
Has been edited to:

public Recipe readRecipe(String recipeId) {
		if (Integer.parseInt(recipeId) > 0)
			return this.recipes.read(recipeId);
		else
			throw new RuntimeException("Invalid id");
	}